src/main/java/com/matt_richardson/gocd/websocket_notifier/PipelineDetailsPopulator.java

```diff
@@ -12,15 +12,20 @@
 import java.net.HttpURLConnection;
 import java.net.URL;
 
+import org.apache.commons.codec.binary.Base64;
+
 public class PipelineDetailsPopulator {
     private static Logger LOGGER = Logger.getLoggerFor(GoNotificationPlugin.class);
+    private static PluginConfig pluginConfig;
     private int httpPort;
 
     public PipelineDetailsPopulator() {
-        this.httpPort = 8153;
+        pluginConfig = new PluginConfig();
+        this.httpPort = pluginConfig.getGoHttpPort();
     }
 
     public PipelineDetailsPopulator(int httpPort) {
+        pluginConfig = new PluginConfig();
```

Comment:
Not sure I like `new`ing up a new instance here - it should be passed in as a parameter, using the instance passed down from WebSocketPipelineListener. This could replace the `int httpPort` param.